packages/logger/src/server-logger.ts

```diff
@@ -0,0 +1,87 @@
+import { createLogger, format, transports, Logger as WinstonLogger } from "winston";
+import winstonRotate from "winston-daily-rotate-file";
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+import fs from 'fs';
+
+// Get current directory
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+// Set up log directory
+const logDirectory = `${__dirname}/logs`;
+
+if (!fs.existsSync(logDirectory)) {
+  fs.mkdirSync(logDirectory);
+  console.log('Logs folder created');
+}
+
+
+export default class Logger {
+  private static instance: Logger;
+  private logger: WinstonLogger;
+  private logLevel: string;
+  private logFilePrefix: string;
+
+  private constructor(logLevel: string = "info", logFilePrefix: string = "plane-log") {
```

Comment:
convert those log level and log file prefix as entities inside an object. 

Basically constructor will take a object, you destruct it and use it. 

By doing this like and server logger can be initialised with same type of object.